pkg/apis/serving/v1beta1/component.go

```diff
@@ -44,7 +44,7 @@ const (
 
 // Constants
 var (
-	SupportedStorageURIPrefixList = []string{"gs://", "s3://", "pvc://", "file://", "https://", "http://"}
+	SupportedStorageURIPrefixList = []string{"gs://", "s3://", "pvc://", "file://", "https://", "http://", "secret://"}
```

Comment:
Why do we need to introduce this artificial scheme type? Can we construct back the s3 or other url from the storage spec?

Reply:
At the KServe controller level, the controller may not know the bucket name because the `default-bucket` in the storage API is optional. Unless the controller reads the secret in the user namespace to get the default-bucket value for every model deployment.

Reply:
that's why I strongly suggest removing `default-bucket` and have user explicitly specify the bucket on the path
```
spec:
  predictor:
    sklearn:
      storage:
        key: localMinIO
        path: example-models/sklearn. # path including the bucket
```

Reply:
So I think @njhill has this requirement where the default-bucket is a parameter in the secret. @njhill what do you think?

Reply:
I have the `secret://` uri scheme because the controller doesn't know what type of storages are in the secret. To know the type of storage, we either need to let the controller pull the user secret every time a new model is deployed/updated or ask them to provide the storageUri that has the storage type.

Reply:
Seems like the credentials package already listing all the secrets in the user namespace. We can have add some extra logics to also check the storage `type` fields in the common secret to know the storage type in controller time if that's okay with everyone.